ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ImportsManager.java

```diff
@@ -51,7 +51,8 @@
     // Regex patterns
     private static final Pattern CLONEABLE_SIGNATURE_PATTERN =
             Pattern.compile("readonly\\|xml<[^>]*>\\|\\(Cloneable\\)\\[]\\|map<Cloneable>\\|table<map<Cloneable>>");
-    private static final Pattern FULLY_QUALIFIED_MODULE_ID_PATTERN = Pattern.compile("([\\w]+)/([\\w.]+):([\\d.]+):");
+    private static final Pattern FULLY_QUALIFIED_MODULE_ID_PATTERN =
```

Comment:
Shall we create an issue to track adding tests with the kind of version that caused issues?

Reply:
There is an issue #28038 to test external library support. or shall I create another issue?

Reply:
Let's create another one. This is something we can try and test by introducing our own project, right?

Reply:
Done #29649